repo/blob/azure/azure_versioned_test.go

```diff
@@ -0,0 +1,218 @@
+package azure_test
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/internal/clock"
+	"github.com/kopia/kopia/internal/gather"
+	"github.com/kopia/kopia/internal/testlogging"
+	"github.com/kopia/kopia/internal/testutil"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/azure"
+)
+
+func TestGetBlobVersions(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be with Immutable Storage with Versioning enabled
```

Comment:
This should be asserted in the test (after getting the environment and creating the bucket) and fail the test if the container does not have the required properties?

Reply:
If versioning is not enabled it'll fail on `st, err = azure.New` with `cannot create point-in-time view for non-versioned container`